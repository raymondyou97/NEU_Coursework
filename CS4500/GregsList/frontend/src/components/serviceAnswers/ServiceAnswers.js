import React from 'react';
import '../../styles/serviceQA.css';

class ServiceAnswers extends React.Component {
  componentDidMount() {
    this.props.fetchServiceAnswers();
  }

  render() {
    const {
      serviceAnswers,
      selectedServiceAnswer,
      serviceAnswerIDChoices,
      handleTrueFalseChange,
      handleMinRangeAnswer,
      handleMaxRangeAnswer,
      handleChoiceAnswer,
    } = this.props;

    return (
      <div className="service-answer-table">
        <h3>Service Answers</h3>
        <table className="service-q-a-table">
          <thead className="service-q-a-header">
            {/* This is the name of the columns */}
            <tr className="service-q-a-table-row">
              <th>ID</th>
              <th>True or False</th>
              <th>Min Range Answer</th>
              <th>Max Range Answer</th>
              <th>Choice Answer</th>
              <th>Commands</th>
            </tr>
          </thead>
          <tbody>
            {/* This is the input fields for the columns */}
            <tr className="service-q-a-input-row">
              <td>{serviceAnswerIDChoices}</td>
              <td>
                <input
                  type="text"
                  onChange={handleTrueFalseChange}
                  placeholder="t/f"
                  value={selectedServiceAnswer.trueFalseAnswer}
                  className="service-answer-input-true-false"
                />
              </td>
              <td>
                <input
                  type="number"
                  onChange={handleMinRangeAnswer}
                  placeholder="0"
                  value={selectedServiceAnswer.minRangeAnswer}
                  className="service-answer-input-min-range"
                />
              </td>
              <td>
                <input
                  type="number"
                  onChange={handleMaxRangeAnswer}
                  placeholder="0"
                  value={selectedServiceAnswer.maxRangeAnswer}
                  className="service-answer-input-max-range"
                />
              </td>
              <td>
                <input
                  type="number"
                  onChange={handleChoiceAnswer}
                  placeholder="0"
                  value={selectedServiceAnswer.choiceAnswer}
                  className="service-answer-input-choice"
                />
              </td>
              <td className="service-q-a-input-buttons">
                <button
                  className="service-q-a-add-button"
                  onClick={this.props.createServiceAnswer}
                >
                  Add
                </button>
                <button
                  className="service-q-a-update-button"
                  onClick={this.props.updateServiceAnswer}
                >
                  Update
                </button>
              </td>
            </tr>
            {serviceAnswers.map(serviceAnswer => {
              return (
                <tr key={serviceAnswer.id} className="service-q-a-table-row">
                  <td>{serviceAnswer.id}</td>
                  {serviceAnswer.trueFalseAnswer ? (
                    <td>{serviceAnswer.trueFalseAnswer.toString()}</td>
                  ) : (
                    <td>-</td>
                  )}
                  {serviceAnswer.minRangeAnswer ? (
                    <td>{serviceAnswer.minRangeAnswer}</td>
                  ) : (
                    <td>-</td>
                  )}
                  {serviceAnswer.maxRangeAnswer ? (
                    <td>{serviceAnswer.maxRangeAnswer}</td>
                  ) : (
                    <td>-</td>
                  )}
                  {serviceAnswer.choiceAnswer ? (
                    <td>{serviceAnswer.choiceAnswer}</td>
                  ) : (
                    <td>-</td>
                  )}
                  <td>
                    <button
                      className="service-q-a-show-button"
                      onClick={() =>
                        this.props.showServiceAnswerDetails(serviceAnswer.id)
                      }
                    >
                      Show
                    </button>
                    <button
                      className="service-q-a-delete-button"
                      onClick={() =>
                        this.props.deleteServiceAnswer(serviceAnswer.id)
                      }
                    >
                      Delete
                    </button>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
        <div className="pagination-row">
          <select onChange={this.props.handleCountChange}>
            <option value={10}>10</option>
            <option value={25}>25</option>
            <option value={50}>50</option>
            <option value={100}>100</option>
            <option value={'All'}>All</option>
          </select>
          <button className="pagination-button" onClick={this.props.pageLeft}>
            Prev
          </button>
          <button className="pagination-button" onClick={this.props.pageRight}>
            Next
          </button>
        </div>
      </div>
    );
  }
}

export default ServiceAnswers;
