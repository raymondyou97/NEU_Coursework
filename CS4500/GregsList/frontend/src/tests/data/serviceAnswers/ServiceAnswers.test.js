'use es6';

import React from 'react';
import { shallow } from 'enzyme';
import ServiceAnswers from '../../../components/serviceAnswers/ServiceAnswers';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SERVICE_ANSWER_LIST } from './ServiceAnswerData';

configure({ adapter: new Adapter() });

test('Snapshot renders', () => {
  let selectedServiceAnswer = {
    trueFalseAnswer: '',
    minRangeAnswer: '',
    maxRangeAnswer: '',
    choiceAnswer: '',
  };

  const component = renderer.create(
    <ServiceAnswers
      serviceAnswers={SERVICE_ANSWER_LIST}
      selectedServiceAnswer={selectedServiceAnswer}
      fetchServiceAnswers={jest.fn()}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot renders when there is date entered into the input boxes', () => {
  let selectedServiceAnswer = {
    trueFalseAnswer: false,
    minRangeAnswer: 1,
    maxRangeAnswer: 3,
    choiceAnswer: 2,
  };

  const component = renderer.create(
    <ServiceAnswers
      serviceAnswers={SERVICE_ANSWER_LIST}
      selectedServiceAnswer={selectedServiceAnswer}
      fetchServiceAnswers={jest.fn()}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test(
  'Initial DOM contains correct number of rows of service answers' +
    ' including input row',
  () => {
    let selectedServiceAnswer = {
      trueFalseAnswer: '',
      minRangeAnswer: '',
      maxRangeAnswer: '',
      choiceAnswer: '',
    };

    let component = shallow(
      <ServiceAnswers
        serviceAnswers={[]}
        selectedServiceAnswer={selectedServiceAnswer}
        fetchServiceAnswers={jest.fn()}
      />
    );

    expect(
      component.find('.service-answer-table .service-q-a-table-row').length
    ).toEqual(1);

    component = shallow(
      <ServiceAnswers
        serviceAnswers={SERVICE_ANSWER_LIST}
        selectedServiceAnswer={selectedServiceAnswer}
        fetchServiceAnswers={jest.fn()}
      />
    );

    expect(
      component.find('.service-answer-table .service-q-a-table-row').length
    ).toEqual(4);
  }
);

test('Service answer component input value changes if input is entered', () => {
  let selectedServiceAnswer = {
    trueFalseAnswer: false,
    minRangeAnswer: 1,
    maxRangeAnswer: 3,
    choiceAnswer: 2,
  };

  let component = shallow(
    <ServiceAnswers
      serviceAnswers={SERVICE_ANSWER_LIST}
      selectedServiceAnswer={selectedServiceAnswer}
      fetchServiceAnswers={jest.fn()}
    />
  );

  expect(
    component.find('.service-answer-input-true-false').props().value
  ).toEqual(false);
  expect(
    component.find('.service-answer-input-min-range').props().value
  ).toEqual(1);
  expect(
    component.find('.service-answer-input-max-range').props().value
  ).toEqual(3);
  expect(component.find('.service-answer-input-choice').props().value).toEqual(
    2
  );

  selectedServiceAnswer = {
    trueFalseAnswer: true,
    minRangeAnswer: null,
    maxRangeAnswer: null,
    choiceAnswer: null,
  };

  component = shallow(
    <ServiceAnswers
      serviceAnswers={SERVICE_ANSWER_LIST}
      selectedServiceAnswer={selectedServiceAnswer}
      fetchServiceAnswers={jest.fn()}
    />
  );

  expect(
    component.find('.service-answer-input-true-false').props().value
  ).toEqual(true);
  expect(
    component.find('.service-answer-input-min-range').props().value
  ).toEqual(null);
  expect(
    component.find('.service-answer-input-max-range').props().value
  ).toEqual(null);
  expect(component.find('.service-answer-input-choice').props().value).toEqual(
    null
  );
});
