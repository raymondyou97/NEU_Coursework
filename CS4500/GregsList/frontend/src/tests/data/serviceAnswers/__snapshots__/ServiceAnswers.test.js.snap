// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div
  className="service-answer-table"
>
  <h3>
    Service Answers
  </h3>
  <table
    className="service-q-a-table"
  >
    <thead
      className="service-q-a-header"
    >
      <tr
        className="service-q-a-table-row"
      >
        <th>
          ID
        </th>
        <th>
          True or False
        </th>
        <th>
          Min Range Answer
        </th>
        <th>
          Max Range Answer
        </th>
        <th>
          Choice Answer
        </th>
        <th>
          Commands
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="service-q-a-input-row"
      >
        <td />
        <td>
          <input
            className="service-answer-input-true-false"
            placeholder="t/f"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="service-answer-input-min-range"
            placeholder="0"
            type="number"
            value=""
          />
        </td>
        <td>
          <input
            className="service-answer-input-max-range"
            placeholder="0"
            type="number"
            value=""
          />
        </td>
        <td>
          <input
            className="service-answer-input-choice"
            placeholder="0"
            type="number"
            value=""
          />
        </td>
        <td
          className="service-q-a-input-buttons"
        >
          <button
            className="service-q-a-add-button"
          >
            Add
          </button>
          <button
            className="service-q-a-update-button"
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          1
        </td>
        <td>
          true
        </td>
        <td />
        <td />
        <td />
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          2
        </td>
        <td>
          false
        </td>
        <td>
          1
        </td>
        <td>
          3
        </td>
        <td>
          2
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          3
        </td>
        <td>
          true
        </td>
        <td>
          50
        </td>
        <td>
          100
        </td>
        <td>
          51
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="pagination-row"
  >
    <select>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={25}
      >
        25
      </option>
      <option
        value={50}
      >
        50
      </option>
      <option
        value={100}
      >
        100
      </option>
      <option
        value="All"
      >
        All
      </option>
    </select>
    <button
      className="pagination-button"
    >
      Prev
    </button>
    <button
      className="pagination-button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Snapshot renders when there is date entered into the input boxes 1`] = `
<div
  className="service-answer-table"
>
  <h3>
    Service Answers
  </h3>
  <table
    className="service-q-a-table"
  >
    <thead
      className="service-q-a-header"
    >
      <tr
        className="service-q-a-table-row"
      >
        <th>
          ID
        </th>
        <th>
          True or False
        </th>
        <th>
          Min Range Answer
        </th>
        <th>
          Max Range Answer
        </th>
        <th>
          Choice Answer
        </th>
        <th>
          Commands
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="service-q-a-input-row"
      >
        <td />
        <td>
          <input
            className="service-answer-input-true-false"
            placeholder="t/f"
            type="text"
            value={false}
          />
        </td>
        <td>
          <input
            className="service-answer-input-min-range"
            placeholder="0"
            type="number"
            value={1}
          />
        </td>
        <td>
          <input
            className="service-answer-input-max-range"
            placeholder="0"
            type="number"
            value={3}
          />
        </td>
        <td>
          <input
            className="service-answer-input-choice"
            placeholder="0"
            type="number"
            value={2}
          />
        </td>
        <td
          className="service-q-a-input-buttons"
        >
          <button
            className="service-q-a-add-button"
          >
            Add
          </button>
          <button
            className="service-q-a-update-button"
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          1
        </td>
        <td>
          true
        </td>
        <td />
        <td />
        <td />
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          2
        </td>
        <td>
          false
        </td>
        <td>
          1
        </td>
        <td>
          3
        </td>
        <td>
          2
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr
        className="service-q-a-table-row"
      >
        <td>
          3
        </td>
        <td>
          true
        </td>
        <td>
          50
        </td>
        <td>
          100
        </td>
        <td>
          51
        </td>
        <td>
          <button
            className="service-q-a-show-button"
            onClick={[Function]}
          >
            Show
          </button>
          <button
            className="service-q-a-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="pagination-row"
  >
    <select>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={25}
      >
        25
      </option>
      <option
        value={50}
      >
        50
      </option>
      <option
        value={100}
      >
        100
      </option>
      <option
        value="All"
      >
        All
      </option>
    </select>
    <button
      className="pagination-button"
    >
      Prev
    </button>
    <button
      className="pagination-button"
    >
      Next
    </button>
  </div>
</div>
`;
